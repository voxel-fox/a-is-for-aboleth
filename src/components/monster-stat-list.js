import * as PropTypes from "prop-types";
import React from "react";
import { css } from "react-emotion";
import statMod from "../utils/stat-mod";

const statRowStyle = css`
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 0 0.5rem;
`;

class MonsterStatList extends React.Component {
  static propTypes = {
    data: PropTypes.arrayOf(
      PropTypes.shape({
        attr: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired
      })
    ).isRequired
  };

  render() {
    const { data } = this.props;

    const StatRow = ({ stat: { value, label, attr } }) => {
      return (
        <div className={statRowStyle}>
          <dt>
            <abbr title={attr}>{label}</abbr>
          </dt>
          <dd
            className={css`
              margin: 0;
              font-size: 1.2em;
            `}
          >
            {value}
          </dd>
          <dd
            className={css`
              margin: 0;
            `}
          >
            {statMod(value)}
          </dd>
        </div>
      );
    };

    return (
      <dl
        className={css`
          display: flex;
          width: 300px;
          justify-content: center;
        `}
      >
        {data.map(attr => (
          <StatRow key={attr.label} stat={attr} />
        ))}
      </dl>
    );
  }
}

export default MonsterStatList;
