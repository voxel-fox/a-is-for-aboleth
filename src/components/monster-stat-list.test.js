import { StatRow, MonsterStatList } from "./monster-stat-list";

describe("<StatRow />", () => {
  const stat = {
    attr: "Strength",
    label: "STR",
    value: 11
  };

  it("renders correctly", () => {
    const wrapper = shallow(<StatRow stat={stat} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<MonsterStatList />", () => {
  const data = [
    {
      attr: "Strength",
      label: "STR",
      value: 11
    },
    {
      attr: "Dexterity",
      label: "DEX",
      value: 13
    },
    {
      attr: "Wisdom",
      label: "WIS",
      value: 8
    }
  ];

  it("renders correctly", () => {
    const wrapper = shallow(<MonsterStatList data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
