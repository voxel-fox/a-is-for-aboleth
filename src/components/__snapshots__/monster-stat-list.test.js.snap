// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonsterStatList /> renders correctly 1`] = `
<dl
  css={
    Object {
      "name": "1m0avwd-statListStyle",
      "styles": "display:flex;width:300px;justify-content:center;label:statListStyle;",
    }
  }
>
  <StatRow
    key="STR"
    stat={
      Object {
        "attr": "Strength",
        "label": "STR",
        "value": 11,
      }
    }
  />
  <StatRow
    key="DEX"
    stat={
      Object {
        "attr": "Dexterity",
        "label": "DEX",
        "value": 13,
      }
    }
  />
  <StatRow
    key="WIS"
    stat={
      Object {
        "attr": "Wisdom",
        "label": "WIS",
        "value": 8,
      }
    }
  />
</dl>
`;

exports[`<StatRow /> renders correctly 1`] = `
<div
  css={
    Object {
      "name": "sary25-statRowStyle",
      "styles": "display:flex;align-items:center;flex-direction:column;margin:0 0.5rem;label:statRowStyle;",
    }
  }
>
  <dt>
    <abbr
      title="Strength"
    >
      STR
    </abbr>
  </dt>
  <dd
    css={
      Object {
        "name": "1vpzcww-statValueStyle",
        "styles": "margin:0;font-size:1.2em;label:statValueStyle;",
      }
    }
  >
    11
  </dd>
  <dd
    css={
      Object {
        "name": "rtzr43-statModStyle",
        "styles": "margin:0;label:statModStyle;",
      }
    }
  >
    0
  </dd>
</div>
`;
